### PR TITLE
INT-B-21456-shipment-email-modification-for-retiree-separatee-counselor-approved-move-details

### PR DESCRIPTION
**Pull request by samaysofo**
**NOTE: Here is a doc to the the screenshots** [B-21456.docx](https://github.com/user-attachments/files/17368157/B-21456.docx)
## Summary
Currently when a  Counselor submits a move details , emails that gets sent to retirees and separatees state that their move is scheduled to deliver to their HOS/HOR/PLEAD instead of the delivery address on their shipment. So these changes are to let the delivery address on the email to display the right address.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [x] Has the branch been pulled in and checked out?
- [x] Have the BL acceptance criteria been met for this change?
- [x] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

**Local testing for emails, you will need to start the server with this:**
EMAIL_BACKEND=ses AWS_REGION=us-gov-west-1  aws-vault exec transcom-gov-dev -- make server_run

### How to test

1. Access the application
2. Login as a Customer
3. Create a new move (Select Retirement/ Separation) on the "Tell us about your move orders page".
4. Go through the process of Setting  up a shipments (e.g HHG/PPM) to create a shipment , sign a complete(Make sure to copy/save your move code for later). 
5. Login as a Service Counselor(SC)
6. Search for your move code in step 4 and navigate to it.
7. Make sure you edit/fix any required fields.
8. Click on "Submit move details" and click "Yes, submit" button on the modal.
9. You will receive email with subject"YOur counselor has approved your move details" in the email you provided during the Move/shipment creation process.
10. Note that the verbiage in the email "This is a confirmation that your counselor has approved move details for the assigned move code [YOUR_MOVE_CODE] to [**Destination_address_you_provided_for_shipment**].

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
image
image
image
image
image
image

### CODE REVIEW

**Comment by samaysofo** [comment]
@ryan-mchugh i updated branch name.

**Review by deandreJones — APPROVED** [review]

**Review by msaki-caci — APPROVED** [review]
LGTM but it does seem that you have some failures in the Circle tests

**Comment by samaysofo** [comment]
The pr has been updated and all circleci tests are passing now, holding off merge for a few hrs if anyone else would like to take a look at it again.

File: pkg/notifications/move_counseled.go [hunk 1]
```diff
@@ -66,15 +67,28 @@ func (m MoveCounseled) emails(appCtx appcontext.AppContext) ([]emailContent, err
 		originDutyLocationName = &originDSTransportInfo.Name
 	}
 
+	destinationAddress := orders.NewDutyLocation.Name
+	isSeparateeOrRetireeOrder := orders.OrdersType == internalmessages.OrdersTypeRETIREMENT || orders.OrdersType == internalmessages.OrdersTypeSEPARATION
+	if isSeparateeOrRetireeOrder && len(move.MTOShipments) > 0 && move.MTOShipments[0].DestinationAddress != nil {
+		streetAddr1, streetAddr2, streetAddr3 := *move.MTOShipments[0].DestinationAddress, "", ""
```
Comment (ryan-mchugh) [inline pkg/notifications/move_counseled.go:73]: (minor) var naming is misleading for `streetAddr1` here, it's actually the destinationAddress

**Pull request merged**